.github/workflows/release.yml

```diff
@@ -109,20 +110,43 @@ jobs:
         with:
           version: ${{ inputs.version }}
           tag_message_template: 'Release ${VERSION}'
+          # Don't push tag, we'll do that after merging up
+          push_tag: false
 
       - name: "Bump to next development release and commit"
         uses: mongodb-labs/drivers-github-tools/bump-version@v2
         with:
           version: ${{ inputs.version }}
           version_bump_script: "./bin/update-release-version.php to-next-patch-dev"
           commit_template: 'Back to -dev'
+          # Don't push commit, we still need to merge up
+          push_commit: false
+
+      - name: "Determine branch to merge up to"
+        id: get-next-branch
+        uses: alcaeus/automatic-merge-up-action/get-next-branch@main
+        with:
+          ref: ${{ github.ref_name }}
+          branchNamePattern: 'v<major>.<minor>'
+          fallbackBranch: 'master'
 
-      # TODO: Manually merge using ours strategy. This avoids merge-up pull requests being created
-      # Process is:
-      # 1. switch to next branch (according to merge-up action)
-      # 2. merge release branch using --strategy=ours
-      # 3. push next branch
-      # 4. switch back to release branch, then push
+      - name: "Manually merge up changes"
+        if: ${{ steps.get-next-branch.outputs.hasNextBranch }}
+        run: |
+          git checkout ${NEXT_BRANCH}
+          git merge --strategy=ours ${RELEASE_BRANCH}
+          git push origin ${NEXT_BRANCH}
+          git checkout ${RELEASE_BRANCH}
+        env:
+          NEXT_BRANCH: ${{ steps.get-next-branch.outputs.branchName }}
+          RELEASE_BRANCH: ${{ github.ref_name }}
+
+      - name: "Push tag and release branch"
+        run: |
+          git push origin ${RELEASE_BRANCH}
+          git push origin ${{ inputs.version }}
```

Comment:
Alternatively, would `git push origin tag ${{ inputs.version }}` be less ambiguous? I picked that up while reading the [git-push docs](https://git-scm.com/docs/git-push) and https://stackoverflow.com/q/5195859

I also noted the advice in the SO thread to avoid using `--tags`, which we historically did (see: [RELEASING.md](https://github.com/mongodb/mongo-php-driver/blob/master/RELEASING.md#tag-the-release)). It appears that [tag-version/action.yml](https://github.com/mongodb-labs/drivers-github-tools/blob/7edc545fcf00ff1da4910079b7488c81d7c583b0/tag-version/action.yml#L54) is still doing that, so perhaps you'd like to revisit that at some point so it only explicitly pushes one tag.

Reply:
TIL `tag <name>` is a thing. Updated to ensure what we're pushing is a tag ref.